cmd/api/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"github.com/getsentry/sentry-go/gin"
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/viper"
+	"github.com/swaggo/gin-swagger"
+	"github.com/swaggo/gin-swagger/swaggerFiles"
+	"github.com/trustwallet/blockatlas/api"
+	"github.com/trustwallet/blockatlas/config"
+	_ "github.com/trustwallet/blockatlas/docs"
+	"github.com/trustwallet/blockatlas/pkg/ginutils"
+	"github.com/trustwallet/blockatlas/pkg/logger"
+	"github.com/trustwallet/blockatlas/platform"
+	"github.com/trustwallet/blockatlas/storage"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+const (
+	defaultPort       = "8420"
+	defaultConfigPath = "../../config.yml"
+)
+
+var (
+	port, confPath string
+	cache          *storage.Storage
+	sg             gin.HandlerFunc
+)
+
+func init() {
+	sg = sentrygin.New(sentrygin.Options{})
+	cache = storage.New()
+
+	flag.StringVar(&port, "p", defaultPort, "port for api")
+	flag.StringVar(&confPath, "c", defaultConfigPath, "config file for api")
+
+	flag.Parse()
+
+	confPath, err := filepath.Abs(defaultConfigPath)
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	config.LoadConfig(confPath)
+	logger.InitLogger()
+	platform.Init()
+
+	host := viper.GetString("storage.redis")
+	err = cache.Init(host)
+	if err != nil {
+		logger.Fatal(err)
+	}
+}
+
+func main() {
+	gin.SetMode(viper.GetString("gin.mode"))
+	engine := gin.New()
+
+	engine.Use(gin.Recovery())
+	engine.Use(ginutils.CheckReverseProxy, sg)
+	engine.Use(ginutils.CORSMiddleware())
+
+	engine.OPTIONS("/*path", ginutils.CORSMiddleware())
+
+	engine.GET("swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler))
```

Comment:
Did you move the API routes inside the CMD package 😱? CMD needs to be in charge only about the initialization

Reply:
CMD package is not exists right now. CMD is folder to run multiple services. Each of them is a single binary. I have already sent examples about using it like these. Here we init the routes during starting the service

Reply:
https://github.com/golang-standards/project-layout#cmd

Reply:
```
/cmd
Main applications for this project.

The directory name for each application should match the name of the executable you want to have (e.g., /cmd/myapp).

It's common to have a small main function that imports and invokes the code from the /internal and /pkg directories and nothing else.
```

Reply:
So, do you see any opposite of it in this file?

Reply:
there are API routes inside this file

Reply:
This is a good point to think about redesigning blocklatlas/api package. It is not  intuitive with multiple files with weird naming for some of them as will as almost random functions and methods for api

Reply:
can you put this API routes inside another method or package, please?